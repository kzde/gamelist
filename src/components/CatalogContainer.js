import React from 'react';
import styled from 'styled-components';
import Card from './Card';
import { colorPalettes, margins, cardSize, media } from '../utils/styleUtils';

const Wrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(4, ${cardSize.width});
  grid-gap: ${margins.large};
  background: ${colorPalettes.lightGrey};
  padding: ${margins.medium};
  padding-top: 0;
  ${media.desktop`grid-template-columns: repeat(3, ${cardSize.width});`}
  ${media.tablet`grid-template-columns: repeat(2, ${cardSize.width});`}
  ${media.phone`grid-template-columns: repeat(1, ${cardSize.width});`}
`;

const CatalogContainer = ({games}) => {
  return (
    <Wrapper>
      {
        games.map((game, index) => {
          return <Card
            key={`${game.name}${index}`}
            title={game.name}
            subTitle={game.platform}
            cover={game.cover}
          />
        })
      }
    </Wrapper>

  )
}

export default CatalogContainer;