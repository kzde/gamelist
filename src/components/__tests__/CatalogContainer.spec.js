import React from 'react';
import renderer from 'react-test-renderer';
import { catalogContainer as CatalogContainer } from '../CatalogContainer';

jest.mock('../Card', () => 'Card');

const props = {
  games: [
    { name: 'game', platform: 'platform', cover: 'cover' },
    { name: 'another game', platform: 'another platform', cover: 'another cover' }
  ],
  history: {
    push: () => {}
  }
};

describe('CatalogContainer', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<CatalogContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
