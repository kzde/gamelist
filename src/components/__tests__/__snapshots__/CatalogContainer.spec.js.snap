// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogContainer should renders correctly 1`] = `
<div
  className="CatalogContainer__Wrapper-sc-1ewu4fh-0 jTwKqB"
>
  <Card
    cover="cover"
    subTitle="platform"
    title="game"
  />
  <Card
    cover="another cover"
    subTitle="another platform"
    title="another game"
  />
</div>
`;
